docs/examples/201/policy-azmonitor-agent-and-dcr-association/policyDefinition.bicep

```diff
@@ -0,0 +1,325 @@
+            ]
+          }
+        ]
+      }
+      then: {
+        effect: 'deployIfNotExists'
+        details: {
+          type: 'Microsoft.Insights/dataCollectionRuleAssociations'
+          name: dcrAssociationName
+          roleDefinitionIds: [
+            '/providers/microsoft.authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c' // contributor RBAC role for deployIfNotExists effect
+          ]
+          existenceCondition: {
+            allOf: [
+              {
+                field: 'Microsoft.Compute/virtualMachines/extensions/type'
+                equals: 'AzureMonitorWindowsAgent'
+              }
+              {
+                field: 'Microsoft.Compute/virtualMachines/extensions/publisher'
+                equals: 'Microsoft.Azure.Monitor'
+              }
+              {
+                field: 'Microsoft.Compute/virtualMachines/extensions/provisioningState'
+                equals: 'Succeeded'
+              }
+            ]
+          }
+          deployment: {
+            properties: {
+              mode: 'incremental'
+              template: {
+                '$schema': 'http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#'
+                contentVersion: '1.0.0.0'
+                parameters: {
+                  resourceName: {
+                    type: 'String'
+                    metadata: {
+                      displayName: 'resourceName'
+                      description: 'Name of the resource'
+                    }
+                  }
+                  resourcelocation: {
+                    type: 'String'
+                    metadata: {
+                      displayName: 'resourceLocation'
+                      description: 'Location of the resource'
+                    }
+                  }
+                  DCRResourceID: {
+                    type: 'String'
+                    metadata: {
+                      displayName: 'DCRResourceID'
+                      description: 'Resource ID of the Data Collection Rule'
+                    }
+                  }
+                }
+                resources: [
+                  {
+                    name: '[concat(parameters(\'resourceName\'), \'/${vmExtensionName}\')]'
+                    type: 'Microsoft.Compute/virtualMachines/extensions'
+                    location: '[parameters(\'resourceLocation\')]'
+                    apiVersion: '2018-06-01'
+                    properties: {
+                      publisher: vmExtensionPublisher
+                      type: vmExtensionType
+                      typeHandlerVersion: vmExtensionTypeHandlerVersion
+                      autoUpgradeMinorVersion: 'true'
+                    }
+                  }
+                  {
+                    name: '[concat(parameters(\'resourceName\'), \'/Microsoft.Insights/${dcrAssociationName}\')]'
+                    type: 'Microsoft.Compute/virtualMachines/providers/dataCollectionRuleAssociations'
```

Comment:
Use the `scope` property instead of combining types:
```
scope: '[resourceId(\'Microsoft.Compute/virtualMachines\', parameters(\'resourceName\'))]'
type: 'Microsoft.Insights/dataCollectionRuleAssociations'
name: dcrAssociationName
```
name: '[parameters(\'resourceName\')]'

Reply:
I tested adding `scope` property to the deployifnotexists effect (below line 212) and the bicep deployment errored out

```
Deployment failed. Correlation ID: 72bbd50c-3c32-458d-85ec-c902a22fa661. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
        "details": [
          {
            "code": "BadRequest",
            "message": "{\r\n  \"error\": {\r\n    \"code\": \"InvalidPolicyRule\",\r\n    \"message\": \"Failed to parse policy rule: 'Could not find member 'scope' on object 
of type 'IfNotExistsDetails'. Path 'scope'.'.\"\r\n  }\r\n}"
          }
        ]
      }
    ]
  }
}
```

The policy's deployifnotexist effects work in the existing configuration

Reply:
Thanks for the comment - tested this change out and getting an ARM template deployment error for the DCR association. 

The template wants to set the resource ID of the DCR association to this:

**/subscriptions/xxxx-xxxx-xxxx-xxxx-xxxx/resourceGroups/dcr/providers/subscriptions/xxxx-xxxx-xxxx-xxxx-xxxx/resourceGroups/DCR/providers/Microsoft.Compute/virtualMachines/vm1/providers/Microsoft.Insights/dataCollectionRuleAssociations/association1**

Error:

```
{
    "status": "Failed",
    "error": {
        "code": "InvalidResourceNamespace",
        "message": "The resource namespace 'subscriptions' is invalid."
    }
}
```

Changed code:

```
{
    name: dcrAssociationName
    scope: '[resourceId(\'Microsoft.Compute/virtualMachines\', parameters(\'resourceName\'))]'
    type: 'Microsoft.Insights/dataCollectionRuleAssociations'
    location: '[parameters(\'resourceLocation\')]'
    apiVersion: '2019-11-01-preview'
    properties: {
        dataCollectionRuleId: '[parameters(\'DCRResourceID\')]'
    }
}
```

Reply:
That's quite bizarre... If you have a correlationId for the failure, I'd be interested in taking a look; I won't hold up the PR for it though!

Reply:
> That's quite bizarre... If you have a correlationId for the failure, I'd be interested in taking a look; I won't hold up the PR for it though!

Ty sir! Will reproduce the error and grab the correlationId.